src/client.ts

```diff
@@ -128,12 +132,22 @@ export class UXDClient {
     user: PublicKey,
     options: ConfirmOptions,
     payer?: PublicKey
-  ): TransactionInstruction {
-    const userCollateralATA = findATAAddrSync(
-      user,
-      depository.collateralMint
-    )[0];
-    const userQuoteATA = findATAAddrSync(user, depository.quoteMint)[0];
+  ): Promise<TransactionInstruction> {
+    const perpSide = polarity === PnLPolarity.Positive ? 'short' : 'long'; //'sell' : 'buy';
+
+    const [userCollateralATA, userQuoteATA, limitPriceI80F48] =
+      await Promise.all([
+        utils.token.associatedAddress({
```

Comment:
@acamill so basically `findATAAddrSync` is deprecated in favor of ` utils.token.associatedAddress` ?

Reply:
I think it's the new interface in spl token 0.2.0 but I was accessing it through 01.. cheater.

I think we can roll back to the 0.1.8 way, I did in the program.

If I'm not mistaken.

Reply:
PSA: please do not use `utils.token.associatedAddress` from anchor, or any async `findProgramAddress` .
It was originally async in `@solana/web3.js`, then they changed it to be synchronous, kept the async version for backwards compatibility.
TLDR: use `PublicKey.findProgramAddressSync`
I will change it in this branch to use only the sync version, and remove any use of `utils.token.associatedAddress` from anchor